### PR TITLE
Add support for user/password auth

### PR DESCRIPTION
**Pull request by Spider84**

### CODE REVIEW

File: conf/main.yml [hunk 1]
```diff
@@ -6,6 +6,10 @@ socks5:
   port: 1080
   # Socks5 server address
   address: 127.0.0.1
+  # Socks5 server login
+# login: 'login'
```
Comment (heiher) [inline conf/main.yml:10]: Move to 'username'.

File: src/hev-config.c [hunk 2]
```diff
@@ -13,8 +13,11 @@
 #include "hev-logger.h"
 #include "hev-config.h"
 
-static char srv_address[256];
-static char srv_port[8];
+static HevConfigServer srv = {
+    .addr = NULL,
+    .login = NULL,
+    .password = NULL
+};
```
Comment (heiher) [inline src/hev-config.c:20]: No need for explicit initialization.

File: src/hev-config.c [hunk 3]
```diff
@@ -28,6 +31,72 @@ static int read_write_timeout = 60000;
...
+        fprintf (stderr, "Can't found %s.address!\n", sec);
+        return -1;
+    }
+
+    if ((login && !password) || (!login && password)) {
+        fprintf (stderr, "Must be set both login and password %s!\n", sec);
+        return -1;
+    }
+
+    srv->addr = strdup(addr);
```
Comment (heiher) [inline src/hev-config.c:90]: If it run as library (e.g. JNI), I think needs to free. or use byte array and strncpy like before?

**Review by heiher — CHANGES_REQUESTED** [review]
Thanks for your patch.

File: src/hev-config.c (hunk 3 above)
Comment (Spider84) [reply src/hev-config.c:90]: you application never call this method more then once at launch. And this pointer used all time wile application work.
In "library style" let to user to do free() if it needed.

File: src/hev-config.c (hunk 2 above)
Comment (Spider84) [reply src/hev-config.c:20]: For this pointers is checking for NULL in code to determinate is login and password are sets.

Comment (heiher) [reply src/hev-config.c:20]: Yes. The default value of uninitialized global variables is 0. If you initialize even if the value is 0, more file space is required.

File: src/hev-config.c (hunk 3 above)
Comment (heiher) [reply src/hev-config.c:90]: See: https://github.com/heiher/hev-socks5-tproxy/blob/master/src/hev-jni.c#L80

I think the internal configs is invisible to the user.

**Comment by heiher** [comment]
Merged with amend.

Thank you.

**Pull request closed without merging**